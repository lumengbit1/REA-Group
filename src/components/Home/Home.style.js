import styled from 'styled-components';

export const HomePage = styled.div`
  display: flex;
  justify-content: space-between;
  height: 100%;
`;

export const Title = styled.div`
  font-size: 1.875rem;
  font-weight: bold;
  margin: 0.625rem;
`;

export const PropertiesArea = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  padding-top: 5%;
  width: 49%;
  border: 0.0625rem solid black;
  padding: 1.25rem 0;
  box-sizing: border-box;
`;

export const Block = styled.div`
  border: 0.0625rem dashed #000;
  border-radius: 0.625rem;
  padding: 1.25rem;
  background-color: #fafbfa;
  min-width: 14.0625rem;
  min-height: 12.8125rem;
`;
