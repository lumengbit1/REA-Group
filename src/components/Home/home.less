@borderstyle: 1px solid #000;
.header {
    width: 100%;
    height: 100px;
    margin-right: 10px;
    margin-bottom: 10px;
    border: @borderstyle;
}
.homePage {
    display: flex;
    justify-content: space-between;
    height: 100vh;
    .resultsArea {
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
        width: 50%;
        margin-right: 10px;
        padding-top: 5%;
        border: @borderstyle;
    }
    .savedPropertiesArea {
        display: flex;
        flex: 1;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
        padding-top: 5%;
        border: @borderstyle;
    }
    .title {
        font-size: 30px;
        font-weight: bold;
        margin: 10px;
    }
    .block {
        border: 1px dashed #000;
        border-radius: 10px;
        padding: 20px;
        background-color: #fafbfa;
        min-width: 225px;
        min-height: 205px;
    }
}
