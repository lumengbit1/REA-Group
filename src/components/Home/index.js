import React from 'react';
import CSSModules from 'react-css-modules';
import styles from './home.less';
import PropertyList from '../PropertyList';
import Header from '../Header';

function Home() {
    return (
        <div>
            <div styleName="header">
                <Header />
            </div>
            <div styleName="homePage">
                <div styleName="resultsArea">
                    <div styleName="title">Results</div>
                    <div styleName="block">
                        <PropertyList area="results" btnText="Add Property" btnClass="btnresults" />
                    </div>
                </div>
                <div styleName="savedPropertiesArea">
                    <div styleName="title">Saved Properties</div>
                    <div styleName="block">
                        <PropertyList area="saved" btnText="Remove Property" btnClass="btnsaved" />
                    </div>
                </div>
            </div>
        </div>
    );
}
export default CSSModules(Home, styles);
