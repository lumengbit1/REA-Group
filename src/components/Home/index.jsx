import React from 'react';
import NumberFormat from 'react-number-format';
import { useDispatch, useSelector } from 'react-redux';
import { filter_property } from '../../reducers/actions';
import PropertyList from '../PropertyList';
import {
  HomePage,
  PropertiesArea,
  Title,
  Block,
} from './Home.style';

const Home = () => {
  const [inputPrice, setInputPrice] = React.useState();
  const [totalPrice, setTotalPrice] = React.useState(0);
  const dispatch = useDispatch();

  const savedValue = useSelector((state) => state.getIn(['value', 'saved']));

  const formatter = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
  });

  React.useEffect(() => {
    if (savedValue.isEmpty()) {
      setTotalPrice(0);
    }
    savedValue.forEach((item) => setTotalPrice(totalPrice + Number(item.get('price').slice(1).split(',').join(''))));
  }, [savedValue]);

  return (
    <div>
      <NumberFormat
        thousandSeparator
        prefix="$"
        onValueChange={(values) => setInputPrice(values.value)}
      />
      <button
        type="button"
        onClick={() => dispatch(filter_property(inputPrice))}
      >
        filter
      </button>
      <div>
        {formatter.format(totalPrice)}
      </div>
      <HomePage>
        <PropertiesArea>
          <Title>
            Results
          </Title>

          <Block>
            <PropertyList type="results" />
          </Block>
        </PropertiesArea>

        <PropertiesArea>
          <Title>
            Saved Properties
          </Title>

          <Block>
            <PropertyList type="saved" />
          </Block>
        </PropertiesArea>
      </HomePage>
    </div>
  );
};

export default Home;
