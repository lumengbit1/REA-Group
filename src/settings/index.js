export default {
  LOCATIONIQ_BASE_API_DOMAIN: 'https://eu1.locationiq.com/v1/search.php',
  OPEN_WEATHER_MAP_BASE_API_DOMAIN: 'https://api.openweathermap.org/data/2.5/onecall',
  OPEN_WEATHER_MAP_KEY: '3ad086fd12d0afd2bb6e33faffbea059',
  LOCATIONIQ_KEY: 'pk.2b73038ba628901801e4e1e93d258e7a',
};
