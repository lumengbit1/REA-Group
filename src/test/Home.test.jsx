import React from 'react';
import 'jest-styled-components';
import axios from 'axios';
import * as redux from 'react-redux';
import { Provider } from 'react-redux';
import MockAdapter from 'axios-mock-adapter';
import thunk from 'redux-thunk';
import { render, waitFor, fireEvent } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { combineReducers } from 'redux-immutable';
import Home from '../components/Home';
import eventList from '../reducers/reducer';
import settings from '../settings';

const store = createStore(combineReducers({ value: eventList }), applyMiddleware(thunk));

const mockSavedData = [
  {
    price: '$526,500',
    agency: {
      brandingColors: {
        primary: '#000000',
      },
      logo: 'http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif',
    },
    id: '4',
    mainImage:
      'http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg',
  },
];

describe('Render Test', () => {
  it('case: expect rendering correct', () => {
    const { container } = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});

describe('filter function test', () => {
  it('1: expect filter button onclick event', async () => {
    const mock = new MockAdapter(axios);
    mock.onGet(`${settings.SAVED_BASE_API_DOMAIN}`).reply(200, mockSavedData);

    const useDispatchSpy = jest.spyOn(redux, 'useDispatch');
    const mockDispatchFn = jest.fn();

    useDispatchSpy.mockReturnValue(mockDispatchFn);

    const { getByTestId } = render(
      <redux.Provider store={store}>
        <Home />
      </redux.Provider>,
    );
    await waitFor(() => getByTestId('filter'));
    fireEvent.click(getByTestId('filter'));
    expect(mockDispatchFn).toBeCalled();
  });

  it('2: expect input change event', async () => {
    const mock = new MockAdapter(axios);
    mock.onGet(`${settings.SAVED_BASE_API_DOMAIN}`).reply(200, mockSavedData);

    const useDispatchSpy = jest.spyOn(redux, 'useDispatch');
    const mockDispatchFn = jest.fn();

    useDispatchSpy.mockReturnValue(mockDispatchFn);

    const { getByTestId } = render(
      <redux.Provider store={store}>
        <Home />
      </redux.Provider>,
    );

    await waitFor(() => getByTestId('input'));
    fireEvent.change(getByTestId('input'), { target: { value: '23' } });
    expect(getByTestId('input').value).toBe('$23');
  });
});
