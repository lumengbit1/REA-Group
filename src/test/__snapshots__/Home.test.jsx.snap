// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test case: expect rendering correct 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
}

.c2 {
  font-size: 1.875rem;
  font-weight: bold;
  margin: 0.625rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 5%;
  width: 49%;
  border: 0.0625rem solid black;
  padding: 1.25rem 0;
  box-sizing: border-box;
}

.c3 {
  border: 0.0625rem dashed #000;
  border-radius: 0.625rem;
  padding: 1.25rem;
  background-color: #fafbfa;
  min-width: 14.0625rem;
  min-height: 12.8125rem;
}

<div>
  <div>
    <input
      data-testid="input"
      inputmode="numeric"
      type="text"
      value=""
    />
    <button
      data-testid="filter"
      type="button"
    >
      filter
    </button>
    <div
      data-testid="total"
    >
      $0.00
    </div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          Results
        </div>
        <div
          class="c3"
        >
          <div
            data-testid="loading"
          >
            Loading...
          </div>
        </div>
      </div>
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          Saved Properties
        </div>
        <div
          class="c3"
        >
          <div
            data-testid="loading"
          >
            Loading...
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
